lib/template_provider.js

```diff
@@ -24,6 +24,16 @@ const ResourceCache = require('./resource_cache').ResourceCache;
 const Template = require('./template').Template;
 const { FsSchemaProvider } = require('./schema_provider');
 
+/**
+ * Strip the extension off of a file
+ *
+ * @param {string} fileName - the file name
+ * @returns {string}
+ */
+function _stripExtension(fileName) {
```

Comment:
Any specific reason why this is a separate func and is "private", or could this be moved under `BaseTemplateProvider`?

Reply:
I see now it's also being used by `GitHubSchemaProvider`.
So my comments below will be kind of a nit :)

In that case, maybe move it under a util or at least remove the `_` since it's being exported?

Reply:
Updated. I was already thinking I should just make a `utils`. It's "private" for now (internal API not exposed in index.js).